### PR TITLE
Check validity of input manifests

### PR DESCRIPTION
**Pull request by bwester**
Adds a check that validates the input data of a manifest when it is being read.
Invalid manifests will be rejected.

Currently, the validity checks only verify that required fields are provided.

### CODE REVIEW

File: pkg/pods/manifest.go [hunk 1]
```diff
@@ -320,3 +324,22 @@ func (m manifest) SignatureData() (plaintext, signature []byte) {
...
+// data is inconsistent or "nil" otherwise.
+func ValidManifest(m Manifest) error {
+	if m.ID() == "" {
+		return fmt.Errorf("manifest must contain an 'id'")
+	}
+	for key, stanza := range m.GetLaunchableStanzas() {
+		switch {
+		case stanza.LaunchableType == "":
+			return fmt.Errorf("'%s': launchable must contain a 'launchable_type'", key)
+		case stanza.LaunchableId == "":
```
Comment (anthonybishopric) [inline pkg/pods/manifest.go]: I would like to remove `LaunchableId` and favor the use of the map key. Perhaps unrelated to this change, but something I'd like to ensure works later.

Comment (bwester) [inline pkg/pods/manifest.go]: Currently, the key is ignored and `LaunchableId` is used everywhere.  Once that changes, we can alter the validation routine here.

Comment (anthonybishopric) [inline pkg/pods/manifest.go]: :+1:

**Comment by anthonybishopric** [comment]
:ship:

**Pull request merged**